messaging/src/test/java/org/axonframework/tracing/NestingSpanFactoryTest.java

```diff
@@ -0,0 +1,115 @@
+
+class NestingSpanFactoryTest {
+
+    private final Clock clock = Mockito.spy(Clock.class);
+    private final SpanFactory delegate = Mockito.mock(SpanFactory.class);
+    private final Span mockSpan = Mockito.mock(Span.class);
+    private final NestingSpanFactory spanFactory = NestingSpanFactory.builder()
+                                                                     .clock(clock)
+                                                                     .delegate(delegate)
+                                                                     .build();
+    private final Supplier<String> nameSupplier = () -> "spanName";
+    EventMessage<Object> message = GenericEventMessage.asEventMessage("payload");
+
+    @BeforeEach
+    void setUp() {
+        when(clock.instant()).thenReturn(Instant.now());
+    }
+
+    @Test
+    void createsNestedSpanIfRecentMessage() {
+
+        spanFactory.createLinkedHandlerSpan(nameSupplier, message);
+
+        verify(delegate).createHandlerSpan(nameSupplier, message, true);
+    }
+
+    @Test
+    void createsLinkedSpanIfOldMessage() {
+        EventMessage<Object> payload = GenericEventMessage.asEventMessage("payload");
+        when(clock.instant()).thenReturn(Instant.now().plusSeconds(121));
+
+        Supplier<String> nameSupplier = () -> "spanName";
+        spanFactory.createLinkedHandlerSpan(nameSupplier, payload);
+
+        verify(delegate).createHandlerSpan(nameSupplier, payload, false);
+    }
+
+    @Test
+    void rootTracesCreatedWillDelegate() {
+        when(delegate.createRootTrace(any())).thenReturn(mockSpan);
+
+        spanFactory.createRootTrace(nameSupplier);
+        verify(delegate).createRootTrace(nameSupplier);
+    }
+
+    @Test
+    void dispatchSpansCreatedWillDelegate() {
+        spanFactory.createDispatchSpan(nameSupplier, message);
+
+        Mockito.verify(delegate).createDispatchSpan(nameSupplier, message);
+    }
+
+    @Test
+    void internalSpansCreatedWillDelegate() {
+        spanFactory.createInternalSpan(nameSupplier);
+
+        Mockito.verify(delegate).createInternalSpan(nameSupplier);
+    }
+
+    @Test
+    void internalSpansWithMessageCreatedWillDelegate() {
+        spanFactory.createInternalSpan(nameSupplier, message);
+
+        Mockito.verify(delegate).createInternalSpan(nameSupplier, message);
+    }
+
+    @Test
+    void registerSpanAttributeProviderWillDelegate() {
+        SpanAttributesProvider provider = mock(SpanAttributesProvider.class);
+        spanFactory.registerSpanAttributeProvider(provider);
+
+        Mockito.verify(delegate).registerSpanAttributeProvider(provider);
+    }
+
+    @Test
+    void propagateContextDelegate() {
+        Message original = mock(Message.class);
+        Message modified = mock(Message.class);
+
+        when(delegate.propagateContext(original)).thenReturn(modified);
+
+        Message result = spanFactory.propagateContext(original);
+        assertSame(result, modified);
+    }
+}
```

Comment:
Nit: no tests for the builder methods? ;-) Wouldn't be the first time there's a bug in the builders...

Reply:
Adding them!